### PR TITLE
[T] Move services to root

### PR DESCRIPTION
**Pull request by viniciuskneves**
## Proposal

*For now you can ignore CI and so on*

- Moved services to root folder
  - **From**: \<rootDir\>/src/services
  - **To**: \<rootDir\>/services
- New way of importing services, so you don't need to fully import the service anymore
  - **From**
```javascript
import { DateService } from 'homeday-blocks';

const { generateDateCycles } = DateService; // or just use DateService.generateDateCycles
```
  - **To**
```javascript
import { generateDateCycles } from 'homeday-blocks/services/date';
```

### Advantage
- We can import just the function we need/want so we don't import the full service

### Disadvantage
- Some refactoring + breaking change
- [Mess(i)](https://sportbild.bild.de/fotos-skaliert/wird-gegen-inter-fehlen-barca-superstar-lionel-messi-201307831-66582448/2,w=993,c=0.sport.jpg) folder structure as `services` won't live inside `src` anymore
  - Regarding this we can think about moving other stuffs outside the `src` folder. We could have `homeday-blocks/components/form/HdInput` for example but that is not the goal of this PR (open for discussion though)

### CODE REVIEW

File: src/stories/HdCalendar.stories.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 /* eslint-disable import/no-extraneous-dependencies */
 import { storiesOf } from '@storybook/vue';
 import { action } from '@storybook/addon-actions';
-import { generateDateCycles } from 'homeday-blocks/src/services/date';
-
+import { generateDateCycles } from 'homeday-blocks/services/date';
```
Comment (viniciuskneves) [inline src/stories/HdCalendar.stories.js]: Take a look in here.

File: tests/unit/services/date.spec.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import {
   getDaysDateRange,
   getIntlDateString,
   generateDateCycles,
-} from '@/services/date';
+} from 'homeday-blocks/services/date';
```
Comment (viniciuskneves) [inline tests/unit/services/date.spec.js]: Take a look in here.

**Comment by SinisaG** [comment]
I'm not sure advantages outweigh disadvantages in this case. I am specifically reserved about breaking the folder structure. 

I think services are not so big that they would be a serious overhead. We could instead:

- Rethink if services are grouped correctly? Can we break them down further to avoid potential bloating? 
- You can still access part of the service, just instead of doing `import { generateDateCycles } from 'homeday-blocks/services/date';`, you have `src` there as well. As this is "not usual", I think that is ok. So side affect here is smaller, in my opinion, then sacrificing folder structure of the project. More our library grows, more the folder structure will be important to keep it sane.

Open for discussion on this though.

**Comment by viniciuskneves** [comment]
@homeday-de/frontend-engineering I would appreciate your input in here.

**Comment by aym3nb** [comment]
In my opinion, I think even the advantage itself is a disadvantage because we need to know the absolute path of the service in order to import it. so I think the current way is cleaner.

**Comment by viniciuskneves** [comment]
> Just an idea.
> What about exporting everything on main.js and then have `import { generateDateCycles } from 'homeday-blocks';` ?

I think it is too broad this way. I think we won't move this one forward. Gonna close it for now.

**Pull request closed without merging**